src/Data/Units.purs

```diff
@@ -285,16 +285,9 @@ baseRepresentation du
 
     -- Replace "gram" by "kilo gram"
     replace u =
-      case runDerivedUnit u of
-        (b : Nil) →
-          let b' =
-                case b.baseUnit of
-                  BaseUnit rec →
-                    if rec.long == "gram"
-                      then b { baseUnit = BaseUnit (rec { long = "kilogram"
-                                                        , short = "kg" }) }
-                      else b
-          in DerivedUnit (singleton b')
+      case u of
+        DerivedUnit (b@{ baseUnit: BaseUnit rec@{ long: "gram" } } : Nil) →
+          DerivedUnit (b { baseUnit = BaseUnit (rec { long = "kilogram", short = "kg" }) } : Nil)
```

Comment:
I condensed this quite a bit, so pleasee check I haven't made a mistake.

Reply:
Looks good - very nice! Didn't realize that pattern matching on records was this powerful.